src/geometry/euclidean/basic.lean

```diff
@@ -64,22 +64,23 @@ conveniently be developed in terms of vectors and then used to deduce
 corresponding results for Euclidean affine spaces.
 -/
 
-variables {V : Type*} [inner_product_space V]
+variables {V : Type*} [inner_product_space ℝ V]
+local notation `⟪`x`, `y`⟫` := @inner ℝ V _ x y
```

Comment:
If you want to use such notation outside the file where the inner product is defined, I think it should be defined in a single place in the `inner_product` file, using `localized` so that files using it then just need to do `open_locale inner_product_real` or whatever rather than redeclaring the notation themselves.

Reply:
The issue with this is that sometimes Lean has trouble figuring out which field is being used when using just the bare notation `inner x y` (this is the case, for example, in the lines you flagged below where I changed the notation just for the first instance of it). The other option would be to declare separate notation for the real and complex case, but most likely in any given application you just want to do one case and never have to worry about this again, so it seems to me like adding a line like this at the top will be nicer.

Reply:
My suggestion is that the line at the top should be of the form `open_locale inner_product_real` rather than something involving `@`, so that anything that cares about the details of implicit arguments to `inner` goes with the definition of `inner` rather than its uses. So that could be the same notation for the real and complex cases, but under different choices of `open_locale`.

Reply:
Ah, I see! Yes that sounds like a good idea.